code/modules/persistence/storage/smartfridge.dm

```diff
@@ -51,14 +51,7 @@
 
 		// Delete some stacks if we want
 		if(stacks_go_missing)
-			var/fuzzy = rand(-5,5)
-			switch(count / max_amount)
-				if(0 to 1)
-					count -= 10+fuzzy // 1 stack or less, lose 10
-				if(1 to 10)
-					count -= max_amount+fuzzy // 1 to 10 stacks, lose a stack
-				if(10 to INFINITY)
-					count -= max_amount*3+fuzzy // 10+ stacks, lose 3 stacks
+			count -= count*0.2 // flat loss of 20%
```

Comment:
Firstly, this will end up setting fractional amounts on stacks. It should be at the very least rounded to a whole number.

The problem with this math is that you'll never remove it all and you'll end up in zeno's paradox land.

<google-sheets-html-origin><!--td {border: 1px solid #ccc;}br {mso-data-placement:same-cell;}-->
[Current] | [This PR] | Also won't work | Also won't work
-- | -- | -- | --
Minus 10 | Minus *0.2 | Minus round(0.2) | Minus floor(0.2)
50 | 50 | 50 | 50
40 | 40 | 40 | 40
30 | 32 | 32 | 32
20 | 25.6 | 26 | 26
10 | 20.48 | 21 | 21
0 | 16.384 | 17 | 17
&nbsp; | 13.1072 | 14 | 14
&nbsp; | 10.48576 | 11 | 12
&nbsp; | 8.388608 | 9 | 10
&nbsp; | 6.7108864 | 7 | 8
&nbsp; | 5.36870912 | 6 | 7
&nbsp; | 4.294967296 | 5 | 6
&nbsp; | 3.435973837 | 4 | 5
&nbsp; | 2.748779069 | 3 | 4
&nbsp; | 2.199023256 | 2 | 4
&nbsp; | 1.759218604 | 2 | 4
&nbsp; | 1.407374884 | 2 | 4
&nbsp; | 1.125899907 | 2 | 4
&nbsp; | 0.9007199255 | 2 | 4
&nbsp; | 0.7205759404 | 2 | 4
&nbsp; | 0.5764607523 | 2 | 4
&nbsp; | 0.4611686018 | 2 | 4
&nbsp; | 0.3689348815 | 2 | 4
&nbsp; | 0.2951479052 | 2 | 4
&nbsp; | 0.2361183241 | 2 | 4
&nbsp; | 0.1888946593 | 2 | 4
&nbsp; | 0.1511157275 | 2 | 4
&nbsp; | 0.120892582 | 2 | 4
&nbsp; | 0.09671406557 | 2 | 4
&nbsp; | 0.07737125246 | 2 | 4
&nbsp; | 0.06189700196 | 2 | 4

</google-sheets-html-origin>

Reply:
Good points.  I think setting count to 0.8 * itself and then taking the floor of that should work though.  That would make it go 50,40,32,25,20,16,12,9,7,5,4,3,2,1,0.  And if we want to keep the randomness we could have the multiplier vary from 0.75 to 0.85 or something like that.

Reply:
Floor(count*0.8) would reach 0, but it would take more than twice as many shifts to reach 0 as the current implementation (on average). If the concern stated is the focus ('people can metagame it to keep sheets longer') then more than doubling the length of persistence seems at best tangential and at worst contrary to that goal.

Reply:
Good point about the whole rounding thing.
The fact they could keep sheets longer wasn't really as much my issue.
It was more that "People felt like they had to follow weird logic in order to do stuff the best way."